src/liballoc/boxed.rs

```diff
@@ -816,6 +816,26 @@ impl From<Box<str>> for Box<[u8]> {
     }
 }
 
+#[stable(feature = "box_from_array", since = "1.44.0")]
+impl<T, const N: usize> From<[T; N]> for Box<[T]>
+where
+    [T; N]: LengthAtMost32,
+{
+    /// Converts a `[T; N]` into a `Box<[T]>`
+    ///
+    /// This conversion allocates on the heap
+    /// and performs a copy of `slice`.
+    ///
+    /// # Examples
+    /// ```rust
+    /// let boxed: Box<[u8]> = Box::from([4, 2]);
+    /// println!("{:?}", boxed);
+    /// ```
+    fn from(slice: [T; N]) -> Box<[T]> {
+        box slice
```

Comment:
`slice` should be called `array`.

Reply:
Do we use the term `array` anywhere? I rarely seen that term used in rust. What is the difference? Is it just because slice have a length?

Reply:
Yip, "array" is actually used quite often. It's [a primitive type](https://doc.rust-lang.org/std/primitive.array.html). "Slice" refers to `[T]`. The length of arrays is known at compile time, the length of slices is not and thus has to be stored at runtime somehow.